### PR TITLE
Extract `exit_error` and `expect_env` functions

### PR DESCRIPTION
**Pull request by mkolesnik**
This logic repeats all over the place, extract small util functions to
cut down repetitiveness and do it properly.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: scripts/shared/compile.sh [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
 
 [[ -n "${BUILD_UPX}" ]] || BUILD_UPX=true
 
 ## Process command line arguments ##
 
-if [[ $# != 2 ]]; then
-    echo "The binary and source must be sepcified!"
-    exit 1
-fi
+[[ $# -eq 2 ]] || exit_error "The binary and source must be sepcified!"
```
Comment (skitt) [inline scripts/shared/compile.sh]: Might as well fix the spelling ;-)

```suggestion
[[ $# -eq 2 ]] || exit_error "The binary and source must be specified!"
```

File: scripts/shared/lib/deploy_funcs [hunk 2]
```diff
@@ -139,10 +132,7 @@ function load_deploytool() {
 function find_submariner_namespace() {
     local namespace
     namespace="$(kubectl get pods --all-namespaces | awk '/submariner/{ print $1 }' | grep -v broker | head -n 1)"
-    if [[ "${namespace}" == "" ]]; then
-        echo "Could not find a submariner deployment namespace" >&2
-        exit 1
-    fi
+    [[ -n "${namespace}" ]] || exit_error "Could not find a submariner deployment namespace"
```
Comment (skitt) [inline scripts/shared/lib/deploy_funcs]:
```suggestion
    [[ -n "${namespace}" ]] || exit_error "Could not find a Submariner deployment namespace"
```

File: scripts/shared/lib/utils [hunk 3]
```diff
@@ -23,6 +23,20 @@ function kubectl() {
     fi
 }
 
+# Exit while printing an error message
```
Comment (skitt) [inline scripts/shared/lib/utils]:
```suggestion
# Exit after printing an error message
```

File: scripts/shared/lib/utils [hunk 4]
```diff
@@ -23,6 +23,20 @@ function kubectl() {
...
+function exit_error() {
+    >&2 echo -e "ERROR: $*"
```
Comment (skitt) [inline scripts/shared/lib/utils]:
```suggestion
    >&2 printf "ERROR: %s\n" "$*"
```

File: test/scripts/compile/test.sh [hunk 5]
```diff
@@ -1,26 +1,20 @@
...
+    [[ "$expected" == "$actual" ]] || exit_error "Expected ${expected@Q}, but got ${actual@Q}"
 }
 
 function test_compile_arch() {
     local binary=$1
     local arch=$2
     ${SCRIPTS_DIR}/compile.sh $binary hello.go
     if ! file $binary | grep -q $arch; then
-        fail "Shoul'dve compiled ${arch@Q} but got $(file $binary)."
+        exit_error "Shoul'dve compiled ${arch@Q} but got $(file $binary)."
```
Comment (skitt) [inline test/scripts/compile/test.sh]:
```suggestion
        exit_error "Should have compiled ${arch@Q} but got $(file $binary)."
```

**Review by skitt — CHANGES_REQUESTED** [review]
A few drive-by spelling fixes, and use `printf` rather than `echo`.

**Review by skitt — APPROVED** [review]

**Comment by mkolesnik** [comment]
Let's make sure to "squash & merge" as I used GH's apply suggestions

**Comment by dfarrell07** [comment]
> Let's make sure to "squash & merge" as I used GH's apply suggestions

Good test for this job, and yes it's properly gating.

> **Linting / No "Apply suggestions from code review" Commits (pull_request) ** Failing

**Review by dfarrell07 — APPROVED** [review]
Nice simplications, thanks Mike!

**Comment by dfarrell07** [comment]
Humm, because the squash commit is only created by the GitHub UI after tests, we can't use that method to get the `Linting / No "Apply suggestions from code review" Commits` job passing.

**Comment by skitt** [comment]
> Humm, because the squash commit is only created by the GitHub UI after tests, we can't use that method to get the `Linting / No "Apply suggestions from code review" Commits` job passing.

Yes, the only way to handle this is for the submitter to squash locally and push. I changed the job to “Required” after Mike asked for a squash, because in my experience on Submariner such requests are pretty much guaranteed to result in an unsquashed rebase.

**Pull request merged**